src/app/tools/point_shapes.h

```diff
@@ -75,6 +79,54 @@ class BrushPointShape : public PointShape {
       m_secondaryColor = loop->getSecondaryColor();
     }
     m_lastGradientValue = -1;
+    m_symmetryMode = (loop->getSymmetry()) ? loop->getSymmetry()->mode() : gen::SymmetryMode::NONE;
+  }
+
+  void setSymmetricCommpressedImages(const Brush* brush) {
+    Image* image = brush->image();
+    m_compressedImages[0].reset(new CompressedImage(image,
+                                                    brush->maskBitmap(),
+                                                    false));
+    switch(m_symmetryMode) {
+      case gen::SymmetryMode::HORIZONTAL:
+      case gen::SymmetryMode::VERTICAL:
+      {
+        doc::algorithm::FlipType flip =
+          (m_symmetryMode == gen::SymmetryMode::HORIZONTAL)? doc::algorithm::FlipType::FlipHorizontal :
+                                                             doc::algorithm::FlipType::FlipVertical;
+        std::unique_ptr<Image> tempImage(Image::create(image->pixelFormat(), image->width(), image->height()));
+        copy_image(tempImage.get(), image);
+        doc::algorithm::flip_image(tempImage.get(), image->bounds(), flip);
+        m_compressedImages[int(m_symmetryMode)].reset(new CompressedImage(tempImage.get(),
+                                                                          brush->maskBitmap(),
+                                                                          false));
+        break;
+      }
+
+      case gen::SymmetryMode::BOTH:
+      {
```

Comment:
Minor change in style:
```suggestion
      case gen::SymmetryMode::BOTH: {
```